src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
-    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value otherwise
+    /// @param serializedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value
+    /// otherwise
     virtual CHIP_ERROR SerializeAdd(EndpointId endpoint,
                                     const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
                                     MutableByteSpan & serializedBytes) override
     {
-        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair;
-        TLV::TLVWriter writer;
-        TLV::TLVType outer;
-        size_t pairTotal  = 0;
-        uint8_t pairCount = 0;
+        app::Clusters::Scenes::Structs::AttributeValuePair::Type aVPairs[kMaxAvPair];
 
+        size_t pairTotal = 0;
         // Verify size of list
         ReturnErrorOnFailure(extensionFieldSet.attributeValueList.ComputeSize(&pairTotal));
-        VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+        VerifyOrReturnError(pairTotal <= ArraySize(aVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
 
+        uint8_t pairCount  = 0;
         auto pair_iterator = extensionFieldSet.attributeValueList.begin();
         while (pair_iterator.Next())
         {
-            aVPair                             = pair_iterator.GetValue();
-            mAVPairs[pairCount].attributeID    = aVPair.attributeID;
-            mAVPairs[pairCount].attributeValue = aVPair.attributeValue;
+            app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair = pair_iterator.GetValue();
+            aVPairs[pairCount].attributeID                                           = aVPair.attributeID;
+            aVPairs[pairCount].attributeValue                                        = aVPair.attributeValue;
             pairCount++;
         }
         ReturnErrorOnFailure(pair_iterator.GetStatus());
-
-        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList;
-        attributeValueList = mAVPairs;
+        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList(aVPairs);
         attributeValueList.reduce_size(pairCount);
 
-        writer.Init(serializedBytes);
-        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer));
-        ReturnErrorOnFailure(app::DataModel::Encode(
-            writer, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList),
-            attributeValueList));
-        ReturnErrorOnFailure(writer.EndContainer(outer));
-
-        return CHIP_NO_ERROR;
+        return EncodeAttributeValueList(attributeValueList, serializedBytes);
```

Comment:
So this used to not update the size of `serializedBytes`, right?  Was that just a bug?

Reply:
Yes, that was a bug that led each EFS to always take the maximum allowed size.